src/utils/atoms/combineAtoms.ts

```diff
@@ -13,3 +14,58 @@ export function combineAtoms<T extends AtomsMap>(
     }, {} as { [key in keyof T]: T[key] extends Atom<infer S> ? S : unknown });
   });
 }
+
+type AsyncAtomsMap = { [key: string]: AsyncAnyAtom<AtomBinded<any>> };
+type CombinedAsyncAtomsState<T> = {
+  loading: boolean;
+  error:
+    | {
+        [key in keyof T]: T[key] extends AsyncAnyAtom<infer P, infer F>
+          ? AtomState<T[key]>['error']
+          : unknown;
+      }
+    | null;
+  data: {
+    [key in keyof T]: T[key] extends AsyncAnyAtom<infer P, infer F>
+      ? AtomState<T[key]>['data']
+      : unknown;
+  };
+  lastParams: {
+    [key in keyof T]: T[key] extends AsyncAnyAtom<infer P, infer F>
+      ? AtomState<T[key]>['lastParams']
+      : unknown;
+  };
+  dirty: true;
+};
+
+export function combineAsyncAtoms<T extends AsyncAtomsMap>(shape: T) {
+  return createAtom(shape, ({ get }): CombinedAsyncAtomsState<T> => {
+    return Object.keys(shape).reduce(
+      (acc, key: keyof T) => {
+        const { loading, error, data, lastParams } = get(key) as AsyncAtomState<
+          unknown,
+          Record<string, unknown>
+        >;
+        if (loading) acc.loading = true;
+        if (error) {
+          if (acc.error === null) {
+            // @ts-ignore
```

Comment:
ts ignores are expected in this file, right?

Reply:
unfortunately I'm not found a way to fix that without ts-ignore. If you now how, please told me